sqrl-server/sqrl-server-vertx/src/main/java/com/datasqrl/graphql/VertxJdbcClient.java

```diff
@@ -1,30 +1,57 @@
 package com.datasqrl.graphql;
 
+import com.datasqrl.graphql.jdbc.PreparedSqrlQueryImpl;
 import com.datasqrl.graphql.server.Context;
 import com.datasqrl.graphql.server.JdbcClient;
 import com.datasqrl.graphql.server.RootGraphqlModel.JdbcQuery;
 import com.datasqrl.graphql.server.RootGraphqlModel.PreparedSqrlQuery;
 import com.datasqrl.graphql.server.RootGraphqlModel.ResolvedJdbcQuery;
 import com.datasqrl.graphql.server.RootGraphqlModel.ResolvedQuery;
+import io.vertx.core.Future;
 import io.vertx.sqlclient.PreparedQuery;
 import io.vertx.sqlclient.Row;
 import io.vertx.sqlclient.RowSet;
 import io.vertx.sqlclient.SqlClient;
+import io.vertx.sqlclient.Tuple;
+import java.util.Map;
 import lombok.Value;
 
 @Value
 public class VertxJdbcClient implements JdbcClient {
-  SqlClient sqlClient;
+  Map<String, SqlClient> clients;
 
   @Override
-  public ResolvedQuery prepareQuery(JdbcQuery pgQuery, Context context) {
+  public ResolvedQuery prepareQuery(JdbcQuery query, Context context) {
+    SqlClient sqlClient = clients.get(query.getDatabase());
+    if (sqlClient == null) {
+      throw new RuntimeException("Could not find database engine: " + query.getDatabase());
+    }
+
     PreparedQuery<RowSet<Row>> preparedQuery = sqlClient
-        .preparedQuery(pgQuery.getSql());
+        .preparedQuery(query.getSql());
 
-    return new ResolvedJdbcQuery(pgQuery,
+    return new ResolvedJdbcQuery(query,
         new PreparedSqrlQueryImpl(preparedQuery));
   }
 
+  public Future<RowSet<Row>> execute(String database, PreparedQuery query, Tuple tup) {
+    if (database == null) {
+      database = "postgres";
+    }
+    SqlClient sqlClient = clients.get(database);
```

Comment:
This kind of means that we can paralelly support duckdb and postgres as a db. Was it a goal?

Reply:
Yeah